### PR TITLE
feat: add picasso-pictograms package

### PR DESCRIPTION
**Pull request by sashuk**
[FX-3636](https://toptal-core.atlassian.net/browse/FX-3636)

### Description

This pull request adds `@toptal/picasso-pictograms` package along with two initial pictograms.

Plan to split work on [FX-3636](https://toptal-core.atlassian.net/browse/FX-3636):

- first pull request adds the `@toptal/picasso-pictograms` package and two pictograms;
- second pull request adds [all the remaining pictograms from Figma](https://www.figma.com/file/6UPpJrQbP02OqBjtGL6KTr/Product-Pictograms?node-id=20%3A5168&t=qjUeLdbWgGJEFQVC-0).

### How to test

- Check Storybook temploy
- Check out updated `README` file (https://github.com/toptal/picasso/tree/fx-3636-illustration-package-for-picasso)
- Check `yarn generate:components-from-svg` command locally, there should be no diffs for already existing SVG React components
- All checks should pass

### Screenshots

https://user-images.githubusercontent.com/1390758/221236671-b8636325-c085-4fe8-b5ab-eece738e232d.mp4

### Development checks

- [x] Remove the SVGO workaround, research problem
- [x] Add changeset according to [guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md) (if needed)
- [x] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- [x] Make sure that additions and changes on design follow [Toptal's BASE design](https://design.toptal.net/), and it's been already discussed with designers at #-base-core
- [N/A] Annotate all `props` in component with documentation
- [N/A] Create `examples` for component
- [x] Ensure that deployed demo has expected results and good examples
- [x] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).

Screenshot 2023-02-27 at 12 00 14

- [x] Self reviewed
- [N/A] Covered with tests

**Breaking change**

- [N/A] codemod is created and showcased in the changeset
- [N/A] test alpha package of Picasso in StaffPortal

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[FX-3636]: https://toptal-core.atlassian.net/browse/FX-3636?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: bin/generate-components-from-svg [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+echo "Converting svg into React components ..."
+
+# TODO: enabled SVGO breaks output locally for SVGs exported from Figma, so using --no-svgo for pictograms
+
+svgr \
+--out-dir $COMPONENTS_PATH \
+--ext tsx \
+--typescript \
+`if [ $1 != 'icon' ]; then echo "--no-svgo"; fi` \
```
Comment (sashuk) [inline bin/generate-components-from-svg:25]: Workaround associated with the `TODO` item above

File: README.md [hunk 2]
```diff
@@ -109,7 +114,7 @@ After Picasso will be released with your changes you can start using your Icon a
 2. Add the new package to:
 
    - the `paths` key in `/tsconfig.json`
-   - the `alias` key in `/.storybook/webpack.config.js`
+   - the `alias` key in `/.storybook/main.js`
```
Comment (sashuk) [inline README.md:117]: Actualising documentation

File: bin/generate-components-from-svg [hunk 3]
```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env bash
```
Comment (sashuk) [inline bin/generate-components-from-svg:1]: Should be compared with https://github.com/toptal/picasso/blob/master/bin/generate-icons file

File: packages/picasso-pictograms/src/Pictogram/_generatorTemplate.js [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    ),
+  ]
+}
+
+const illustrationTemplate = ({ componentName, jsx }, { tpl }) => {
+  const displayName = `'${componentName}'`
+  const svgElement = jsx.openingElement
+
+  // add `className={cx(classes.root, classes[colorClassName], className)}` to svg root tag
+  decorateWithClassNameProp(svgElement)
```
Comment (sashuk) [inline packages/picasso-pictograms/src/Pictogram/_generatorTemplate.js:38]: Decorators are still needed for pictograms

File: packages/picasso-pictograms/src/Pictogram/_generatorTemplate.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
+const types = require('@babel/types')
+
+/**
+ * Add `{propName}={propValue}` to the svg tag
+ */
+const decorateWithProp = (svgElement, propName, propValue) => {
```
Comment (sashuk) [inline packages/picasso-pictograms/src/Pictogram/_generatorTemplate.js:6]: decorateWithProp is still needed for pictograms to provide `className`, `style`, etc. props (the `decorateWithClassNameProp` was removed)

File: bin/generate-components-from-svg [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+# initialize
+CWD="$(dirname "$0")"
+cd "$CWD/../"
+BASE_PATH=$(pwd)
+
+if [ $1 = 'icon' ]
```
Comment (TomasSlama) [inline bin/generate-components-from-svg:8]: ❤️

File: cypress.config.mjs [hunk 7]
```diff
@@ -76,6 +76,10 @@ export default defineConfig({
               process.cwd(),
               './packages/picasso-provider/dist-package'
             ),
+            '@toptal/picasso-pi': path.resolve(
```
Comment (TomasSlama) [inline cypress.config.mjs:79]:
```suggestion
            '@toptal/picasso-pictograms': path.resolve(
```

I guess it is typo

File: packages/picasso-pictograms/package.json [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+    "prepublishOnly": "if [ -d dist-package ]; then cp ./package.json ./dist-package/package.json; fi"
+  },
+  "bugs": {
+    "url": "https://github.com/toptal/picasso/issues"
+  },
+  "homepage": "https://github.com/toptal/picasso/tree/master/packages/picasso-pictograms#readme",
+  "peerDependencies": {
+    "typescript": "~4.7.0",
+    "@material-ui/core": "4.12.4",
+    "@toptal/picasso": "^31.0.0",
```
Comment (TomasSlama) [inline packages/picasso-pictograms/package.json:28]: It looks to me, that we are using `@toptal/picasso` only in storybook examples, so we can probably move it to `devDependencies`, and I think we don't use `@material-ui/core` anywhere in this repo, so we probably don't need it, or am I missing something? 🤔

File: packages/picasso-pictograms/package.json [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+    "@toptal/picasso-shared": "^11.0.0",
+    "react": ">=16.12.0 < 19.0.0"
+  },
+  "devDependencies": {
+    "storybook-readme": "^5.0.9"
+  },
+  "dependencies": {
+    "classnames": "^2.3.1",
```
Comment (TomasSlama) [inline packages/picasso-pictograms/package.json:36]: it seems it is not used

File: packages/picasso-pictograms/package.json [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+    "@babel/types": "^7.20.7"
```
Comment (TomasSlama) [inline packages/picasso-pictograms/package.json:37]: if we need it, it should be in devDependencies

**Review by TomasSlama — COMMENTED** [review]
left a few small comments, but it looks great!

File: cypress.config.mjs (hunk 7 above)
Comment (sashuk) [reply cypress.config.mjs]: @TomasSlama Fixed, thanks!

File: packages/picasso-pictograms/package.json (hunk 8 above)
Comment (sashuk) [reply packages/picasso-pictograms/package.json]: @TomasSlama Cleaned up, thanks!

File: packages/picasso-pictograms/package.json (hunk 9 above)
Comment (sashuk) [reply packages/picasso-pictograms/package.json]: @TomasSlama Fixed, thanks!

File: packages/picasso-pictograms/package.json (hunk 10 above)
Comment (sashuk) [reply packages/picasso-pictograms/package.json:33]: @TomasSlama Fixed, thanks!

**Review by TomasSlama — APPROVED** [review]
Good job! 👍, checked in temploy

File: .changeset/lucky-humans-repeat.md [hunk 11]
```diff
@@ -0,0 +1,7 @@
+---
+'@toptal/picasso': patch
+---
+
+---
+
+- renamings in the icon generation template
```
Comment (mkrl) [inline .changeset/lucky-humans-repeat.md:7]: I don't think it's necessary to bump Picasso version for this. We use changesets to generate changelogs for distributed packages + version bumps, the change in template generator will not change anything in the published Picasso package.

File: README.md [hunk 12]
```diff
@@ -25,50 +25,55 @@ In order to run storybook you need to execute `yarn start` which will spin up st
...
-
-To add a new Icon to Picasso library please follow these steps:
+| Command                          | Description                                                       |
+| -------------------------------- | ----------------------------------------------------------------- |
+| **build:package**                | Build the packages                                                |
+| **build:storybook**              | Build Storybook as static website                                 |
+| **commit**                       | Interactive conventional commits                                  |
+| **generate:component**           | Generate a new component template                                 |
+| **generate:example**             | Generate a new component component code example                   |
+| **generate:components-from-svg** | [Generate JSX components from svgs](#adding-icons-and-pictograms) |
```
Comment (mkrl) [inline README.md:35]: I'm thinking, maybe it's worth to keep `generate:icons` and `generate:pictograms` here as well? As far as I understand, they can be executed separately.

File: package.json [hunk 13]
```diff
@@ -12,7 +12,9 @@
     "circularity": "madge --circular packages/*/src",
     "generate:component": "davinci-code new component",
     "generate:example": "davinci-code new example",
-    "generate:icons": "./bin/generate-icons",
+    "generate:components-from-svg": "yarn generate:icons && yarn generate:pictograms",
```
Comment (mkrl) [inline package.json:15]:
```suggestion
    "generate:svg-components": "yarn generate:icons && yarn generate:pictograms",
```
Just an idea / small nit: sometimes you don't have autocomplete and typing a long command is more painful :smile:

Comment (sashuk) [reply package.json]: @mkrl Sounds good, added!

File: README.md (hunk 12 above)
Comment (sashuk) [reply README.md]: @mkrl Added, thank you!

File: .changeset/lucky-humans-repeat.md (hunk 11 above)
Comment (sashuk) [reply .changeset/lucky-humans-repeat.md]: @mkrl Removed, thank you!

**Comment by sashuk** [comment]
@toptal-anvil ping reviewers

**Review by mkrl — APPROVED** [review]
Looks good, nice job :+1:

**Pull request merged**